src/services/projectService.ts

```diff
@@ -100,6 +103,48 @@ export default class ProjectService implements IProjectService {
         return (duplicateProjects !== undefined);
     }
 
+    public async deleteTag(project: IProject,
+                           tag: string, currentAsset: IAssetMetadata): Promise<IAssetMetadata> {
+        return await this.updateProjectTags(
+            project, tag, currentAsset, (tags) => tags.filter((t) => t !== tag));
+    }
+
+    public async updateTag(project: IProject, tag: string,
+                           newTag: string, currentAsset: IAssetMetadata): Promise<IAssetMetadata> {
+        return await this.updateProjectTags(
+            project, tag, currentAsset, (tags) => tags.map((t) => (t === tag) ? newTag : t));
+    }
+
+    private async updateProjectTags(
+            project: IProject, tag: string, currentAsset: IAssetMetadata, transformer: (tags: string[]) => string[]) {
+        const assetService = new AssetService(project);
+        const assetKeys = Object.keys(project.assets);
+        for (const assetKey of assetKeys) {
```

Comment:
Let's wrap this loop in a `foreEachAsync`

Reply:
I think we can be a bit more proactive here and only update the assets that are currently in the `Tagged` state.

Reply:
There could also be a use case where if all regions are now untagged we need to remove the region which should cause the asset state to transition back into the `Visited` state.